pkg/controller/seed-controller-manager/kubernetes/resources.go

```diff
@@ -999,3 +1012,8 @@ func hostnameToIPList(ctx context.Context, hostname string) ([]net.IP, error) {
 
 	return ipList, nil
 }
+
+func isCiliumInstalled(client ctrlruntimeclient.Client) bool {
+	_, err := client.RESTMapper().RESTMapping(schema.ParseGroupKind("ciliumnetworkpolicies.cilium.io"), "v2")
+	return err == nil
```

Comment:
One thing that worries me about this is what if it's some sort of a temporary error (e.g. a timeout). How often do we reconcile this? Ideally we would use something like `IsNotFound` to be sure that it really doesn't exist. I'm wondering if it's save to use [`IsNoMatchFound`](https://github.com/kubernetes/apimachinery/blob/fa98d6eaedb4caccd69fc07d90bbb6a1e551f00f/pkg/api/meta/firsthit_restmapper.go#L86-L105) here.

Reply:
I wouldn't be too worried about this because I think the reconcile is triggered fairly frequent (this is the loop for _all_ resources per cluster in the seed) and nothing gets removed.